libraries/AP_BattMonitor/AP_BattMonitor_SMBus.cpp

```diff
@@ -18,6 +38,22 @@ bool AP_BattMonitor_SMBus::read_temp(void)
     return false;
 }
 
+// reads the serial number if it's not already known
+// returns true if the read was successful or the number was already known
+bool AP_BattMonitor_SMBus::read_serial_number(void) {
+    uint16_t data;
+
+    // don't recheck the serial number if we already have it
+    if (_serial_number != 0) {
```

Comment:
Isn't 0 a valid serial number? I thought before that 0 for the parameter default would be better than -1, but then I thought that the battery might have the serial number 0.